metrics/metrics.go

```diff
@@ -18,11 +18,12 @@ func HandleMetrics(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	authHeader := req.Header.Get("Authorization")
+	apiKeyHeader := req.Header.Get("x-api-key")
 	var err error
-	config.Cfg.UserID, err = config.ValidateToken(authHeader)
+	config.Cfg.UserID, err = config.ValidateUser(authHeader, apiKeyHeader)
 	if err != nil {
-		log.Printf("error validating token %v", err)
-		http.Error(w, "error validating token", 500)
+		log.Printf(err.Error())
+		http.Error(w, "error validating user", http.StatusUnauthorized)
```

Comment:
Add an empty return after this - otherwise, the user will get this error and `recordMetrics()` will still get called.